src/DockContainerWidget.cpp

```diff
@@ -541,7 +540,10 @@ void DockContainerWidgetPrivate::appendDockAreas(const QList<CDockAreaWidget*> N
 	DockAreas.append(NewDockAreas);
 	for (auto DockArea : NewDockAreas)
 	{
-		_this->connect(DockArea, SIGNAL(viewToggled(bool)), SLOT(onDockAreaViewToggled(bool)));
+        QObject::connect(DockArea,

```

Comment:
What is the reason for switching to the new connect syntax? While it improves compile time type checking I think the syntax and readibility is worsened - especially if I read this sd::bind statement.

Reply:
The main reason was compile failure on on old versions of Qt (tested on Qt 5.8 MSVC2013). Basically moc did couldn't see the definition of `DockContainerWidgetPrivate` so it was creating a call to an incomplete class.
Regardless, the new syntax brings a TON of advantages
+ It is faster at run time
+ it's safer at compile time
+ doesn't require leaking private parts into the public headers (by declaring Q_PRIVATE_SLOT)

The `std::bind` syntax for private slots was [accepted by Qt itself](https://code.woboq.org/qt5/qtbase/src/corelib/itemmodels/qtransposeproxymodel.cpp.html#208)

Reply:
Ok, I'm convinced and will accept this change. So the std::bind syntax is required if the private slot is declared in the private class, is this right? So that means, it would be possible to remove **Q_PRIVATE_SLOT(d, void onDockAreaViewToggled(bool Visible))** from **CDockContainerWidget**? If yes, please do this.

Reply:
It's possible and I already did remove it